src/functions/memoize.ts

```diff
@@ -0,0 +1,22 @@
+// Import Third-party Dependencies
+import lodash from "lodash";
+
+// Import Internal Dependencies
+import { RC } from "../rc.js";
+
+let memoizedValue: Partial<RC> | null = null;
+
+export type memoizeOptions = Partial<RC>;
+
+export function memoize(payload: memoizeOptions, options = { overwrite: true }) {
+  if (memoizedValue === null || options.overwrite) {
+    memoizedValue = payload;
+  }
+  memoizedValue = lodash.merge({}, memoizedValue, payload);
+}
+
+export function memoized(options = { defaultValue: null }) {
```

Comment:
Same as memoize for `options` argument. Please also write the return type for this function.